### PR TITLE
[AS-190] Add Brands Stream

### PR DESCRIPTION
**Pull request by nickolasclarke**
This adds the [Brands endpoint](https://developer.zendesk.com/api-reference/ticketing/account-configuration/brands/) as a Stream. Local run returns as expected. Below is a sample of the rows.

```
[
  {
    "_sdc_batched_at": "2024-08-01 21:32:09.921465",
    "_sdc_deleted_at": null,
    "_sdc_extracted_at": null,
    "active": true,
    "brand_url": "https://delmarva-support.weavegrid.com",
    "created_at": "2024-06-10 18:16:20",
    "default": false,
    "has_help_center": true,
    "help_center_state": "enabled",
    "host_mapping": "delmarva-support.weavegrid.com",
    "id": 24083652170263,
    "logo": null,
    "name": "WeaveGrid for Delmarva",
    "signature_template": "{{agent.signature}}\r\nHave a good experience? Please leave us a review!",
    "subdomain": "delmarva-weavegrid",
    "ticket_form_ids": "[1500000489742, 1500000532402]",
    "updated_at": "2024-06-24 17:00:20",
    "url": "https://ev-pulse.zendesk.com/api/v2/brands/24083652170263.json"
  },
  {
    "_sdc_batched_at": "2024-08-01 21:32:09.925198",
    "_sdc_deleted_at": null,
    "_sdc_extracted_at": null,
    "active": true,
    "brand_url": "https://pepco-support.weavegrid.com",
    "created_at": "2024-06-10 18:17:09",
    "default": false,
    "has_help_center": true,
    "help_center_state": "enabled",
    "host_mapping": "pepco-support.weavegrid.com",
    "id": 24083666747031,
    "logo": null,
    "name": "WeaveGrid for Pepco",
    "signature_template": "{{agent.signature}}\r\nHave a good experience? Please leave us a review!",
    "subdomain": "pepco-weavegrid",
    "ticket_form_ids": "[1500000489742, 1500000532402]",
    "updated_at": "2024-06-24 17:01:41",
    "url": "https://ev-pulse.zendesk.com/api/v2/brands/24083666747031.json"
  },
  {
    "_sdc_batched_at": "2024-08-01 21:32:09.923125",
    "_sdc_deleted_at": null,
    "_sdc_extracted_at": null,
    "active": true,
    "brand_url": "https://fpl-support.weavegrid.com",
    "created_at": "2024-06-17 18:52:56",
    "default": false,
    "has_help_center": true,
    "help_center_state": "enabled",
    "host_mapping": "fpl-support.weavegrid.com",
    "id": 24239119749655,
    "logo": null,
    "name": "WeaveGrid for FPL",
    "signature_template": "{{agent.signature}}\r\nHave a good experience? Please leave us a review!",
    "subdomain": "fpl-weavegrid",
    "ticket_form_ids": "[1500000489742, 1500000532402]",
    "updated_at": "2024-06-24 17:02:44",
    "url": "https://ev-pulse.zendesk.com/api/v2/brands/24239119749655.json"
  },
  {
    "_sdc_batched_at": "2024-08-01 21:32:09.924187",
    "_sdc_deleted_at": null,
    "_sdc_extracted_at": null,
    "active": true,
    "brand_url": "https://luma-support.weavegrid.com",
    "created_at": "2024-06-18 15:42:43",
    "default": false,
    "has_help_center": true,
    "help_center_state": "restricted",
    "host_mapping": "luma-support.weavegrid.com",
    "id": 24262021157783,
    "logo": null,
    "name": "WeaveGrid for LUMA",
    "signature_template": "{{agent.signature}}\r\nHave a good experience? Please leave us a review!",
    "subdomain": "luma-weavegrid",
    "ticket_form_ids": "[1500000489742, 1500000532402]",
    "updated_at": "2024-06-27 05:56:59",
    "url": "https://ev-pulse.zendesk.com/api/v2/brands/24262021157783.json"
  },
  {
    "_sdc_batched_at": "2024-08-01 21:32:09.924688",
    "_sdc_deleted_at": null,
    "_sdc_extracted_at": null,
    "active": true,
    "brand_url": "https://oru-support.weavegrid.com",
    "created_at": "2024-06-18 15:54:17",
    "default": false,
    "has_help_center": true,
    "help_center_state": "restricted",
    "host_mapping": "oru-support.weavegrid.com",
    "id": 24262418180631,
    "logo": null,
    "name": "WeaveGrid for Orange & Rockland",
    "signature_template": "{{agent.signature}}\r\nHave a good experience? Please leave us a review!",
    "subdomain": "oru-weavegrid",
    "ticket_form_ids": "[1500000489742, 1500000532402]",
    "updated_at": "2024-07-18 18:48:20",
    "url": "https://ev-pulse.zendesk.com/api/v2/brands/24262418180631.json"
  },
  {
    "_sdc_batched_at": "2024-08-01 21:32:09.925714",
    "_sdc_deleted_at": null,
    "_sdc_extracted_at": null,
    "active": true,
    "brand_url": "https://pge-support.weavegrid.com",
    "created_at": "2024-07-09 17:48:29",
    "default": false,
    "has_help_center": true,
    "help_center_state": "restricted",
    "host_mapping": "pge-support.weavegrid.com",
    "id": 24766036499607,
    "logo": null,
    "name": "WeaveGrid for PG&E",
    "signature_template": "{{agent.signature}}\r\nHave a good experience? Please leave us a review!",
    "subdomain": "pge-weavegrid",
    "ticket_form_ids": "[1500000489742, 1500000532402]",
    "updated_at": "2024-07-18 18:47:45",
    "url": "https://ev-pulse.zendesk.com/api/v2/brands/24766036499607.json"
  }
]

```

### CODE REVIEW

File: tap_zendesk/streams.py [hunk 1]
```diff
@@ -377,7 +395,7 @@ def sync(self, state):
         # created_at, but zendesk support confirmed with us that
         # satisfaction_ratings are immutable so that created_at =
         # updated_at
-        #start = bookmark_epoch-1
+        # start = bookmark_epoch-1
```
Comment (nickolasclarke) [inline tap_zendesk/streams.py:398]: auto linting going on. I dont much care.

**Comment by bdewilde** [comment]
Should we not be submitting this PR upstream, to the original repo?

**Comment by nickolasclarke** [comment]
I'll cut a second once it is in main on our fork. Who knows if they'll ever accept it.

File: tap_zendesk/streams.py [hunk 2]
```diff
@@ -133,6 +133,24 @@ def raise_or_log_zenpy_apiexception(schema, stream, e):
...
+
+    def sync(self, state):
+        bookmark = self.get_bookmark(state)
+
+        brands = self.client.brands()
+        for brand in brands:
+            if utils.strptime_with_tz(brand.updated_at) >= bookmark:
+                # NB: We don't trust that the records come back ordered by
+                # updated_at (we've observed out-of-order records),
+                # so we can't save state until we've seen all records
```
Comment (bdewilde) [inline tap_zendesk/streams.py:149]: Is this something you observed, or are you copy-pasting this logic from elsewhere?

Comment (nickolasclarke) [reply tap_zendesk/streams.py:149]: I've copied this logic from one of the same blocks below. This comment shows up 17 times in this file.

**Review by bdewilde — APPROVED** [review]
Trusting that this all works as it should, and not reviewing logic in detail that's mostly a copy-paste of what's already here. Please do submit these changes to the upstream repo!

**Pull request merged**